tests/component/test_timing_properties.py

```diff
@@ -0,0 +1,221 @@
+        task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+        task.timing.cfg_samp_clk_timing(1000, sample_mode=AcquisitionType.CONTINUOUS)
+
+        task.timing.samp_quant_samp_mode = AcquisitionType.FINITE
+
+        assert task.timing.samp_quant_samp_mode == AcquisitionType.FINITE
+
+
+def test__timing__reset_enum_property__returns_default_value(any_x_series_device):
+    """Test for validating reset for enum property."""
+    with nidaqmx.Task() as task:
+        task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+        task.timing.cfg_samp_clk_timing(1000, sample_mode=AcquisitionType.FINITE)
+
+        del task.timing.samp_quant_samp_mode
+
+        assert task.timing.samp_quant_samp_mode == AcquisitionType.CONTINUOUS
+
+
+def test__timing__getter_float64_property__returns_value(any_x_series_device):
+    """Test for validating getter for float64 property."""
+    with nidaqmx.Task() as task:
+        task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+        task.timing.cfg_samp_clk_timing(1000)
+
+        assert task.timing.samp_clk_rate == 1000
+
+
+def test__timing__setter_float64_property__returns_assigned_value(any_x_series_device):
+    """Test for validating setter for float64 property."""
+    with nidaqmx.Task() as task:
+        task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+        task.timing.cfg_samp_clk_timing(1000)
+
+        task.timing.samp_clk_rate = 2000
+
+        assert task.timing.samp_clk_rate == 2000
+
+
+def test__timing__reset_float64_property__returns_default_value(any_x_series_device):
+    """Test for validating reset for float64 property."""
+    with nidaqmx.Task() as task:
+        task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+        default_value = task.timing.samp_clk_rate
+        task.timing.cfg_samp_clk_timing(10000)
+
+        del task.timing.samp_clk_rate
+
+        assert task.timing.samp_clk_rate == default_value
+
+
+def test__timing__getter_uint32_property__returns_value(any_x_series_device):
+    """Test for validating getter for unsigned int32 property."""
+    with nidaqmx.Task() as task:
+        task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+        task.timing.cfg_samp_clk_timing(1000)
+
+        assert task.timing.samp_clk_timebase_div == 100000
+
+
+def test__timing__setter_uint32_property__returns_assigned_value(any_x_series_device):
+    """Test for validating setter for unsigned int32 property."""
+    with nidaqmx.Task() as task:
+        task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+        task.timing.cfg_samp_clk_timing(1000)
+
+        task.timing.samp_clk_timebase_div = 500
+
+        assert task.timing.samp_clk_timebase_div == 500
+
+
+def test__timing__reset_uint32_property__returns_default_value(any_x_series_device):
```

Comment:
@zhindes @bkeryan currently this test is failing. The issue is `samp_clk_timebase_div` is not resetting to its initial/default value.

- `default_value = task.timing.samp_clk_timebase_div` -> got the initial/default value before configuring sample clock
     -  `default_value = task.timing.samp_clk_timebase_div = 100000`
- `task.timing.cfg_samp_clk_timing(500)` -> then, configured sample clock with **sample rate = 500**
    - `task.timing.samp_clk_timebase_div = 200000`
- `del task.timing.samp_clk_timebase_div` -> **I expected `samp_clk_timebase_div` to change to its default value(100000) but it remained as 200000**
   - `task.timing.samp_clk_timebase_div = 200000`

Assert fails, since it is not changing to default value after deletion.
![image](https://user-images.githubusercontent.com/126869186/229191018-60d54cea-29fa-4301-88c0-c3af20ba8524.png)

Can you please if this is the expected behavior of reset  `samp_clk_timebase_div` or not?

Reply:
For properties in a task, DAQmx tracks both the user's desired value and the coerced value set by the DAQmx driver.

`task.timing.cfg_samp_clk_timing(500)` sets the desired value of `samp_clk_rate` to 500, and then querying a property verifies the task, which sets the coerced value of `samp_clk_timebase_div` to 200000 (assuming the default timebase rate is 100e6). 

Resetting `samp_clk_timebase_div` has no effect because it wasn't set by the user. Resetting `samp_clk_rate` and verifying the task will set the coerced value of `samp_clk_timebase_div` to 100000 (assuming the default sample rate is 1000 and the default timebase rate is 100e6).

The solution is to set and reset the same property. If you set both `samp_clk_rate` and `samp_clk_timebase_div`, I think `samp_clk_timebase_div` takes precedence and `samp_clk_rate` is coerced.

Try this:
```
        task.timing.cfg_samp_clk_timing(1000)
        default_value = task.timing.samp_clk_timebase_div
        task.timing.samp_clk_timebase_div = 200000
        assert task.timing.samp_clk_rate == 500

        del task.timing.samp_clk_timebase_div

        assert task.timing.samp_clk_timebase_div == default_value
```

Reply:
Ok got it! Thank you for the clear explanation. Tried the suggested test scenario. Now the test pass as expected. :)